homeassistant/components/sia/alarm_control_panel.py

```diff
@@ -0,0 +1,288 @@
+        if self._unsub:
+            self._unsub()
+        if self._remove_unavailability_tracker:
+            self._remove_unavailability_tracker()
+
+    async def async_handle_event(self, event: Event):
+        """Listen to events for this port and account and update states.
+
+        If the port and account combo receives any message it means it is online and can therefore be set to available.
+        """
+        await self.assume_available()
+        if int(event.data[EVENT_ZONE]) == self._zone:
+            new_state = CODE_CONSEQUENCES.get(event.data[EVENT_CODE])
+            if new_state:
+                self._attr.update(
+                    {
+                        ATTR_LAST_MESSAGE: event.data[EVENT_MESSAGE],
+                        ATTR_LAST_CODE: event.data[EVENT_CODE],
+                        ATTR_LAST_TIMESTAMP: event.data[EVENT_TIMESTAMP],
+                    }
+                )
+                self.state = new_state
+                if not self.registry_entry.disabled:
+                    self.async_schedule_update_ha_state()
+
+    @callback
+    def _schedule_immediate_update(self):
+        self.async_schedule_update_ha_state(True)
+
+    @property
+    def name(self) -> str:
+        """Get Name."""
+        return self._name
+
+    @property
+    def ping_interval(self) -> int:
+        """Get ping_interval."""
+        return str(self._ping_interval)
+
+    @property
+    def state(self) -> str:
+        """Get state."""
+        return self._state
+
+    @property
+    def account(self) -> str:
+        """Return device account."""
+        return self._account
+
+    @property
+    def unique_id(self) -> str:
+        """Get unique_id."""
+        return self._unique_id
+
+    @property
+    def available(self) -> bool:
+        """Get availability."""
+        return self._is_available
+
+    @property
+    def device_state_attributes(self) -> dict:
+        """Return device attributes."""
+        return self._attr
+
+    @property
+    def should_poll(self) -> bool:
+        """Return False if entity pushes its state to HA."""
+        return False
+
+    @state.setter
+    def state(self, state: str):
+        """Set state."""
+        temp = self._old_state if state == PREVIOUS_STATE else state
+        self._old_state = self._state
+        self._state = temp
+
+    async def assume_available(self):
+        """Reset unavalability tracker."""
+        if not self.registry_entry.disabled:
+            await self._async_track_unavailable()
+
+    @callback
+    async def _async_track_unavailable(self) -> bool:
+        """Reset unavailability."""
+        if self._remove_unavailability_tracker:
+            self._remove_unavailability_tracker()
+        self._remove_unavailability_tracker = async_track_point_in_utc_time(
+            self.hass,
+            self._async_set_unavailable,
+            utcnow() + self._ping_interval + PING_INTERVAL_MARGIN,
+        )
```

Comment:
Can this be moved to the pypi package?

Reply:
No, the underlying package is stateless, just waits for TCP calls, so the notion of availability is completely related to HA and its components.